File: go/problems/1_two_sum.go
package main

func twoSum(nums []int, target int) []int {
	hashMap := make(map[int]int)
	for i, num := range nums {
		diff := target - num
		if key, exists := hashMap[diff]; exists {
			return []int{key, i}
		}
		hashMap[num] = i
	}
	return []int{}
}

File: ts/problems/1-two-sum.ts
export function twoSum(nums: Array<number>, target: number): Array<number> {
  const hashMap: Record<number, number> = {};
  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];
    if (Object.hasOwn(hashMap, diff)) {
      return [hashMap[diff], i];
    }
    hashMap[nums[i]] = i;
  }
  return [];
}
